src/test/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/NewApiScannerInstallerTest.java

```diff
@@ -0,0 +1,69 @@
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class NewApiScannerInstallerTest {
+
+    @Test
+    void testActualDownload() throws Exception {
+        String signatureScannerDownloadPath = TestingPropertyKey.TEST_BLACKDUCK_SIGNATURE_SCANNER_DOWNLOAD_PATH.fromEnvironment();
+        String blackDuckUrl = TestingPropertyKey.TEST_BLACK_DUCK_SERVER_URL.fromEnvironment();
+        String blackDuckUsername = TestingPropertyKey.TEST_USERNAME.fromEnvironment();
+        String blackDuckPassword = TestingPropertyKey.TEST_PASSWORD.fromEnvironment();
+        assertTrue(StringUtils.isNotBlank(signatureScannerDownloadPath));
+
+        IntLogger logger = new BufferedIntLogger();
+        BlackDuckServerConfigBuilder blackDuckServerConfigBuilder = BlackDuckServerConfig.newCredentialsBuilder();
+        blackDuckServerConfigBuilder.setUrl(blackDuckUrl);
+        blackDuckServerConfigBuilder.setUsername(blackDuckUsername);
+        blackDuckServerConfigBuilder.setPassword(blackDuckPassword);
+        blackDuckServerConfigBuilder.setTimeoutInSeconds(120);
+        blackDuckServerConfigBuilder.setTrustCert(true);
+        blackDuckServerConfigBuilder.setLogger(logger);
+
+        BlackDuckServerConfig blackDuckServerConfig = blackDuckServerConfigBuilder.build();
+        BlackDuckHttpClient blackDuckHttpClient = blackDuckServerConfig.createBlackDuckHttpClient(logger);
+        BlackDuckServicesFactory blackDuckServicesFactory = blackDuckServerConfig.createBlackDuckServicesFactory(blackDuckHttpClient, logger);
+        BlackDuckRegistrationService blackDuckRegistrationService = blackDuckServicesFactory.createBlackDuckRegistrationService();
+
+        OperatingSystemType operatingSystemType = OperatingSystemType.determineFromSystem();
+        ScanPathsUtility scanPathsUtility = new ScanPathsUtility(logger, IntEnvironmentVariables.includeSystemEnv(), operatingSystemType);
+        CleanupZipExpander cleanupZipExpander = new CleanupZipExpander(logger);
+        File downloadTarget = new File(signatureScannerDownloadPath);
+
+        NewApiScannerInstaller newApiScannerInstaller = new NewApiScannerInstaller(logger, blackDuckHttpClient, cleanupZipExpander, scanPathsUtility, new HttpUrl(blackDuckUrl),
+                operatingSystemType,
+                downloadTarget);
+        newApiScannerInstaller.installOrUpdateScanner();
+
+        ScanPaths scanPaths = scanPathsUtility.searchForScanPaths(downloadTarget);
+        assertTrue(scanPaths.isManagedByLibrary());
+        assertTrue(StringUtils.isNotBlank(scanPaths.getPathToJavaExecutable()));
+        assertTrue(StringUtils.isNotBlank(scanPaths.getPathToOneJar()));
+        assertTrue(StringUtils.isNotBlank(scanPaths.getPathToScanExecutable()));
+        assertTrue(new File(scanPaths.getPathToJavaExecutable()).canExecute());
+        assertTrue(new File(scanPaths.getPathToOneJar()).canExecute());
+        assertTrue(new File(scanPaths.getPathToScanExecutable()).canExecute());
+    }
+
+}
```

Comment:
The test instance we use is on a version that would not invoke the NewApiScannerInstaller. Troubleshooting that while looking into IDETECT-4402